server.js

```diff
@@ -0,0 +1,113 @@
+
+const bcrypt = require('bcryptjs');
+const express = require('express');
+const helmet = require('helmet');
+const jwt = require('jsonwebtoken');
+const secrets = require('./secrets.js');
+
+const server = express();
+server.use(helmet());
+server.use(express.json());
+
+// const restrictedRouter = require('./restricted-router.js');
+// server.use('/api/restricted', tokenCheck, restrictedRouter);
+
+const db = require('./data/dbConfig.js');
+
+server.get('/api/users', tokenCheck, async (req,res) => {
+    try {
+        const result = await db('users');
+        res.status(200).json(result);
+    }
+    catch(err) {
+        res.status(500).json({err});
+    }
+});
+
+server.post('/api/register', async (req,res) => {
+    let user = req.body;
+    if (!user.username || !user.password || !user.department) {
+        res.status(400).json({error:'please provide username/password'});
+    } else {
+        user.password = bcrypt.hashSync(user.password, 8);
+        try {
+            const result = await db('users').insert(user);
+            if (result) {
+                const newUser = await db('users').where({'id':result[0]}).first();
```

Comment:
well done getting the new user back instead of just the id (default return from the insert method), but a couple optimizations I'd make: instead of `result` from storing your first await call, I'd recommend being as specific as possible with naming what comes back, so `id` or `newId` or something similar, just to prevent any confusion if you need to use the variable later on in another fashion or something. you can also make your second await call a little cleaner if you already grabbed exactly what you need for the id from the first call, so you can add `.first()` on the end of the first call to return the id object from inside the array it gets returned in by default, or even use array destructuring on the first result like `const [id] = await db('users').insert(user)` since now you have a reference to the id you wanted to get back in the first place, so now your second call can look _much_ cleaner with `.where({id})`